### PR TITLE
Removing assertions from code and replacing with relevant exceptions

### PR DESCRIPTION
**Pull request by asloobq**
Refactors
- move test method out of UIDOperatorService.java
- moving error message string literals to constants

TODO

- [x] - run E2E tests
- [x] integration deploy and test

### CODE REVIEW

File: src/main/java/com/uid2/operator/store/CloudSyncOptOutStore.java [hunk 1]
```diff
@@ -594,15 +603,20 @@ private OptOutStoreSnapshot updateIndexInternal(IndexUpdateContext iuc) {
...
-                    assert this.iteration == 0;
+                    if (this.iteration != 0) {
+                        final String errorMsg = "Should not load more than 1 partition at a time, unless during service bootstrap. Current iteration " + this.iteration;
+                        // Leaving this as a warning as this condition is true in production
+                        LOGGER.warn(errorMsg);
+                    }
                     return this.processPartitions(iuc);
                 } else {
-                    // array size cannot be a negative value
-                    assert numPartitions == 1;
```
Comment (asloobq) [inline src/main/java/com/uid2/operator/store/CloudSyncOptOutStore.java:605]: this condition is always true

File: src/main/java/com/uid2/operator/store/CloudSyncOptOutStore.java [hunk 2]
```diff
@@ -594,15 +603,20 @@ private OptOutStoreSnapshot updateIndexInternal(IndexUpdateContext iuc) {
...
+                    if (!iuc.getDeltasToRemove().isEmpty()) {
+                        final String errorMsg = "Invalid number of Deltas to remove=" + iuc.getDeltasToRemove().size();
+                        LOGGER.error(errorMsg);
+                        throw new IllegalStateException(errorMsg);
+                    }
                     return this.processDeltas(iuc);
                 } else if (numPartitions > 1) {
-                    // should not load more than 1 partition at a time, unless during service bootstrap
-                    assert this.iteration == 0;
+                    if (this.iteration != 0) {
```
Comment (asloobq) [inline src/main/java/com/uid2/operator/store/CloudSyncOptOutStore.java:613]: @mcollins-ttd FYI

File: pom.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.uid2</groupId>
     <artifactId>uid2-operator</artifactId>
-    <version>5.40.86</version>
+    <version>5.40.87-alpha-110-SNAPSHOT</version>
```
Comment (sunnywu) [inline pom.xml:9]: this should be removed from the final merge?

File: src/test/java/com/uid2/operator/V2RequestUtilTest.java [hunk 4]
```diff
@@ -118,4 +136,28 @@ public void testParseRequestWithMalformedJson() {
...
+        JsonObject jsonBody = new JsonObject(response);
+        when(keyManager.getRefreshKey()).thenReturn(refreshKey);
+        when(refreshKey.getId()).thenReturn(Integer.MAX_VALUE);
+        when(refreshKey.getKeyBytes()).thenReturn(Random.getRandomKeyBytes());
+        try {
+            V2RequestUtil.handleRefreshTokenInResponseBody(jsonBody, keyManager, IdentityScope.UID2);
+            fail("IllegalArgumentException not thrown");
+        } catch (IllegalArgumentException e) {
+            assertEquals("Generated refresh token's length=168 is not equal to=388", e.getMessage());
+        }
```
Comment (mcollins-ttd) [inline src/test/java/com/uid2/operator/V2RequestUtilTest.java:161]: Suggestion:

```suggestion
        IllegalArgumentException e = assertThrowsExactly(
                IllegalArgumentException.class,
                () -> V2RequestUtil.handleRefreshTokenInResponseBody(jsonBody, keyManager, IdentityScope.UID2));
        assertEquals("Generated refresh token's length=168 is not equal to=388", e.getMessage());
```

File: src/main/java/com/uid2/operator/monitoring/StatsCollectorVerticle.java [hunk 5]
```diff
@@ -88,7 +88,9 @@ public void handleMessage(Message message) {
             return;
         }
 
-        assert messageItem != null;
+        if (messageItem == null) {
```
[the remark this answers is not in the document]
Comment (asloobq) [reply src/main/java/com/uid2/operator/monitoring/StatsCollectorVerticle.java:91]: Looks like it won't return a null value. If the input string is null it throws an IllegalArgumentException and Json*Exception for bad json string. I'll remove this check. Thanks

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 6]
```diff
@@ -1363,9 +1353,9 @@ private InputUtil.InputVal getTokenInputV1(RoutingContext rc) {
         return null;
     }
 
-    private boolean checkForInvalidTokenInput(InputUtil.InputVal input, RoutingContext rc) {
```
Comment (sunnywu) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:1366]: good that you renamed checkForInvalidTokenInput as after reading the original codes and what bool it returns, this method name is highly deceptive!

**Review by sunnywu — APPROVED** [review]

**Review by mcollins-ttd — APPROVED** [review]

**Pull request merged**